src/React/Basic.js

```diff
@@ -1,39 +1,101 @@
 
-var React = require('react');
-
-exports.react_ = function(spec) {
-  return React.createClass({
-    getInitialState: function() {
-      return spec.initialState;
-    },
-    componentDidMount: function() {
-      var this_ = this;
-      spec.receiveProps(this.props, this.state, function(newState) {
-        return function() {
-          this_.setState(newState);
-        };
-      });
-    },
-    componentWillReceiveProps: function(newProps) {
-      var this_ = this;
-      spec.receiveProps(newProps, this.state, function(newState) {
-        return function() {
-          this_.setState(newState);
-        };
-      });
-    },
-    render: function() {
-      var this_ = this;
-      return spec.render(this.props, this.state, function(newState) {
-        return function() {
-          this_.setState(newState);
-        };
-      });
-    }
+var React = require("react");
+var Fragment = React.Fragment || "div";
+
+// Class polyfill
+var __extends =
+  (this && this.__extends) ||
+  (function() {
+    var extendStatics =
+      Object.setPrototypeOf ||
+      ({ __proto__: [] } instanceof Array &&
+        function(d, b) {
+          d.__proto__ = b;
+        }) ||
+      function(d, b) {
+        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
+      };
+    return function(d, b) {
+      extendStatics(d, b);
+      function __() {
+        this.constructor = d;
+      }
+      d.prototype =
+        b === null
+          ? Object.create(b)
+          : ((__.prototype = b.prototype), new __());
+    };
+  })();
+
+var __shallowCopy = function(obj) {
+  if (obj == null) return;
+  var result = {};
+  Object.keys(obj).forEach(function(key) {
+    result[key] = obj[key];
   });
+  return result;
+};
+
+exports.component_ = function(spec) {
+  function Component(props) {
+    this.state = spec.initialState;
+    return this;
+  }
+  __extends(Component, React.Component);
+
+  Component.prototype.displayName = spec.displayName;
+
+  Component.prototype.componentDidMount = function() {
+    var this_ = this;
+    spec.receiveProps(this.props, this.state, function(newState) {
+      return function() {
+        this_.setState(newState);
+      };
+    });
+  };
+
+  Component.prototype.componentWillReceiveProps = function(newProps) {
+    var this_ = this;
+    spec.receiveProps(newProps, this.state, function(newState) {
+      return function() {
+        this_.setState(newState);
+      };
+    });
+  };
+
+  Component.prototype.render = function() {
+    var this_ = this;
+    return spec.render(this.props, this.state, function(newState) {
+      return function() {
+        this_.setState(newState);
+      };
+    });
+  };
+
+  return Component;
+};
+
+exports.createElement_ = function(el, attrs) {
+  return React.createElement.apply(
+    null,
+    [el, attrs].concat((attrs && attrs.children) || [])
+  );
+};
+
+exports.createElementKeyed_ = function(el, key, attrs) {
+  var attrsClone = __shallowCopy(attrs) || {};
```

Comment:
Can we not accept a record including a `key` property here?

Reply:
Yes, but that seems harder to type in PS.  Have an idea?

Reply:
I was thinking of this:

```
createElementKeyed
  :: forall props
   . ReactComponent { | props }
  -> { key :: String | props }
  -> JSX
createElementKeyed = runFn2 createElementKeyed_

foreign import createElementKeyed_ :: forall props. Fn2 (ReactComponent { | props }) { key :: String | props } JSX
```

Reply:
Oh right, that makes sense.  We'd also be able to just use `createElement_` for its implementation as well.

Reply:
Never mind, we need both implementations because of the warning about keys.